[Shipping Labels] Include the new woo shipping plugin when checking for shipping labels.

toupper opened the pull request:
<!-- Remember about a good descriptive title. -->

<!-- Id number of the GitHub issue this PR addresses. -->

## Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->
With this PR we allow purchasing shipping labels when the new Woo Shipping plugin is installed.

## Testing instructions
<!-- Step by step testing instructions. When necessary break out individual scenarios that need testing, consider including a checklist for the reviewer to go through. -->

- In your store, install and activate the new Woo Shipping plugin (ask me in case you need the file). Deactivate the legacy WooCommerce Shipping & Tax.
- Go through the Shipping Label Purchase flow and verify that it works succesfully.
- Check that refunding shipping labels works as before.
- Check also that the shipping labels are shown in the order details once they are purchased.

## Screenshots
<!-- Include before and after images or gifs when appropriate. -->
N/A

---
- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

toupper committed:
Rename to communicate the legacy nature of the old shipping plugin

toupper committed:
Add information about feature

toupper committed:
Simplify code

ThomazFB committed:
Merge branch 'trunk' into task/enable-new-woo-shipping-plugin

ThomazFB committed:
Fix incorrect WCShip reference in unit tests

dangermattic commented:
<!--
  0 Errors
  1 Warning: This PR is assigned to the mil...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-ios/milestone/192">18.0</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

ThomazFB reviewed (approved):
Tested all basic scenarios and could verify all of them working. Even the refund one. Turns out the issue on the Android side was a code adjustment within the Plugin check. I only had to push some code changes due to some unit tests failing.

Just one small thing: I added a feature flag on the Android code base in order to allow us to quickly disable the new Shipping plugin support if something goes bad and wait the best moment to release the support. Do you think we should do the same here? I see you added a comment in a Feature Flag definition that could be an intention of achieving this.

toupper commented:
Thanks @ThomazFB for reviewing and fix the unit tests!

> Just one small thing: I added a feature flag on the Android code base in order to allow us to quickly disable the new Shipping plugin support if something goes bad and wait the best moment to release the support. Do you think we should do the same here? I see you added a comment in a Feature Flag definition that could be an intention of achieving this.

Actually, the comment was for the shipping labels onboarding feature flag, as it was referring only to the legacy shipping plugin. Now that we added support for the new one, we would have to include it in the shipping labels onboarding too.
But for our case, even if I see the advantages of a feature flag, I think we can merge this without a feature flag; we tested it, and worked well. That way we don't add more complexity to our code. In case we still need to revert it, we can use this PR to do it.
